### PR TITLE
TMA-291: Exchange JSON for MultiJson

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/375/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 7s
- gooddata-ruby-pronto-code-review-zuul : SKIPPED
- gooddata-ruby-unit-tests-zuul : SKIPPED

**Comment by panjan** [comment]
ok to test

File: lib/gooddata/lcm/actions/create_segment_masters.rb [hunk 1]
```diff
@@ -105,7 +107,7 @@ def call(params)
             segment_in[:timestamp] = Time.now.utc.iso8601
 
             # Show new project
-            params.gdc_logger.info JSON.pretty_generate(project.json)
+            params.gdc_logger.info MultiJson.pretty_generate(project.json)
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/create_segment_masters.rb]: Calls 'params.gdc_logger' 3 times (DuplicateMethodCall)

File: lib/gooddata/lcm/lcm.rb [hunk 2]
```diff
@@ -184,7 +186,7 @@ def transfer_label_types(source_project, targets)
         display_forms.each { |display_form| transfer[display_form['meta']['identifier']] = display_form['content']['type'] }
 
         GoodData.logger.info 'Transferring label types'
-        GoodData.logger.info JSON.pretty_generate(transfer)
+        GoodData.logger.info MultiJson.pretty_generate(transfer)
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm.rb]: Calls 'gooddata.logger' 4 times (DuplicateMethodCall)

File: lib/gooddata/models/blueprint/project_builder.rb [hunk 3]
```diff
@@ -85,9 +85,9 @@ def to_json(options = {})
         eliminate_empty = options[:eliminate_empty] || false
 
         if eliminate_empty
-          JSON.pretty_generate(to_hash.reject { |_k, v| v.is_a?(Enumerable) && v.empty? })
+          MultiJson.pretty_generate(to_hash.reject { |_k, v| v.is_a?(Enumerable) && v.empty? })
```
Comment (yenkins) [inline lib/gooddata/models/blueprint/project_builder.rb]: Has the variable name 'v' (UncommunicativeVariableName)

File: lib/gooddata/models/model.rb [hunk 4]
```diff
@@ -170,7 +170,7 @@ def upload_multiple_data(data, project_blueprint, options = { :client => GoodDat
         begin
           Zip::File.open("#{dir}/upload.zip", Zip::File::CREATE) do |zip|
             # TODO: make sure schema columns match CSV column names
-            zip.get_output_stream('upload_info.json') { |f| f.puts JSON.pretty_generate(manifest) }
+            zip.get_output_stream('upload_info.json') { |f| f.puts MultiJson.pretty_generate(manifest) }
```
Comment (yenkins) [inline lib/gooddata/models/model.rb]: Calls 'multijson.pretty_generate(manifest)' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/model.rb]: Contains iterators nested 2 deep (NestedIterators)

Comment (yenkins) [inline lib/gooddata/models/model.rb]: Has the variable name 'f' (UncommunicativeVariableName)

File: lib/gooddata/models/model.rb [hunk 5]
```diff
@@ -253,8 +253,8 @@ def upload_multiple_data(data, project_blueprint, options = { :client => GoodDat
           m += "Columns that are in csv but shouldn't be there (manifest): #{csv_extra}\n" unless csv_extra.empty?
           m += "Columns in the uploaded csv: #{csv_headers}\n"
           m += "Columns in the manifest: #{manifest_cols}\n"
-          m += "Original message:\n#{JSON.pretty_generate(js)}\n"
-          m += "Manifest used for uploading:\n#{JSON.pretty_generate(manifest)}"
+          m += "Original message:\n#{MultiJson.pretty_generate(js)}\n"
```
Comment (yenkins) [inline lib/gooddata/models/model.rb]: Refers to 'm' more than self (maybe move it to another class?) (FeatureEnvy)

Comment (yenkins) [inline lib/gooddata/models/model.rb]: Has the variable name 'm' (UncommunicativeVariableName)

File: lib/gooddata/models/process.rb [hunk 6]
```diff
@@ -201,7 +202,7 @@ def deploy_from_appstore(path, options = { :client => GoodData.client, :project
           end
 
         if res.keys.first == 'asyncTask'
-          res = JSON.parse(client.poll_on_code(res['asyncTask']['links']['poll'], options.merge(process: false)))
+          res = MultiJson.parse(client.poll_on_code(res['asyncTask']['links']['poll'], options.merge(process: false)))
```
Comment (yenkins) [inline lib/gooddata/models/process.rb]: Refers to 'client' more than self (maybe move it to another class?) (FeatureEnvy)

Comment (yenkins) [inline lib/gooddata/models/process.rb]: Refers to 'options' more than self (maybe move it to another class?) (FeatureEnvy)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/376/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/114/ : SUCCESS in 42s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/424/ : SUCCESS in 27s

**Comment by panjan** [comment]
extended test

File: lib/gooddata/commands/datasets.rb [hunk 7]
```diff
@@ -50,7 +52,7 @@ def describe
         name = extract_option('--name') || ask('Enter the dataset name')
         output = extract_option('--output') || ask('Enter path to the file where to save the model description', :default => "#{name}.json")
         open output, 'w' do |f|
-          f << JSON.pretty_generate(:title => name, :columns => columns) + "\n"
+          f << MultiJson.pretty_generate(:title => name, :columns => columns) + "\n"
```
Comment (panjan) [inline lib/gooddata/commands/datasets.rb]: Please replace calls to `pretty_generate` with `dump(foo, :pretty => true)`.

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/377/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1022/ : FAILURE in 33m 50s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/780/ : UNSTABLE in 3m 39s

**Comment by kubamahnert** [comment]
extended test

File: lib/gooddata/commands/datasets.rb [hunk 8]
```diff
@@ -50,7 +52,7 @@ def describe
         name = extract_option('--name') || ask('Enter the dataset name')
         output = extract_option('--output') || ask('Enter path to the file where to save the model description', :default => "#{name}.json")
         open output, 'w' do |f|
-          f << JSON.pretty_generate(:title => name, :columns => columns) + "\n"
+          f << MultiJson.dump({:title => name, :columns => columns}, :pretty => true) + "\n"
```
Comment (yenkins) [inline lib/gooddata/commands/datasets.rb]: Style/SpaceInsideHashLiteralBraces: Space inside { missing.

Comment (yenkins) [inline lib/gooddata/commands/datasets.rb]: Style/SpaceInsideHashLiteralBraces: Space inside } missing.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/379/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/116/ : FAILURE in 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/426/ : SUCCESS in 28s

**Comment by kubamahnert** [comment]
extended test

**Comment by kubamahnert** [comment]
extended test

File: lib/gooddata/models/model.rb [hunk 9]
```diff
@@ -170,7 +170,7 @@ def upload_multiple_data(data, project_blueprint, options = { :client => GoodDat
         begin
           Zip::File.open("#{dir}/upload.zip", Zip::File::CREATE) do |zip|
             # TODO: make sure schema columns match CSV column names
-            zip.get_output_stream('upload_info.json') { |f| f.puts JSON.pretty_generate(manifest) }
+            zip.get_output_stream('upload_info.json') { |f| f.puts MultiJson.dump(manifest, :pretty => true) }
```
Comment (yenkins) [inline lib/gooddata/models/model.rb]: Calls 'multijson.dump(manifest, :pretty => true)' 2 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/381/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/117/ : FAILURE in 1m 56s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/427/ : SUCCESS in 1m 51s

**Comment by kubamahnert** [comment]
extended test - staging

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/382/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1026/ : FAILURE in 1m 57s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/784/ : SUCCESS in 21m 38s

**Comment by panjan** [comment]
extended test - staging

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/384/ : SUCCESS in 27s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1028/ : SUCCESS in 43m 21s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/786/ : SUCCESS in 22m 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/385/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/120/ : FAILURE in 41s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/430/ : SUCCESS in 31s

**Pull request closed without merging**